opentracing_instrumentation/client_hooks/celery.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import absolute_import
+
+from distutils.version import LooseVersion
+
+import opentracing
+from opentracing.ext import tags
+
+from ..request_context import get_current_span, RequestContextManager
+from ._patcher import Patcher
+
+
+try:
+    import celery
+    from celery.app.task import Task
+    from celery.signals import before_task_publish, task_prerun, task_postrun
+except ImportError:  # pragma: no cover
+    pass
+else:
+    _task_apply_async = Task.apply_async
+    if LooseVersion(celery.__version__) >= LooseVersion('4'):
+        def get_parent_span_context(request):
+            return getattr(request, 'parent_span_context', None)
+    else:
+        def get_parent_span_context(request):
+            return request.headers and request.headers.get(
+                'parent_span_context'
+            )
+
+
+def task_apply_async_wrapper(task, args=None, kwargs=None, **other_kwargs):
+    operation_name = 'Celery:apply_async:{}'.format(task.name)
+    span = opentracing.tracer.start_span(operation_name=operation_name,
+                                         child_of=get_current_span())
+    set_common_tags(span, task, tags.SPAN_KIND_RPC_CLIENT)
+
+    with RequestContextManager(span=span), span:
```

Comment:
As I understand `RequestContextManager`  is deprecated https://github.com/uber-common/opentracing-python-instrumentation/blob/master/opentracing_instrumentation/request_context.py#L52 Is it good idea to use it instead span helpers?

Reply:
I've updated the PR.
Thank you!